### ISSUE TITLE
Marginalize over varying effects

### ISSUE THREAD

**Issue by paul-buerkner** [paul-buerkner/brms#489]
In `predict` and related functions, we currently have the option to either condition on varying effects (i.e. include them in the predictions) or to simply ignore them (i.e. conditioning on zero). Instead of ignoring, we should actually marginalize over them, which is something different in models with non-identity links. In the past, this distinction has (understandably) let to confusion not only for brms users. 

I argue we should add the possibility of marginalizing over varying effects via monte-carlo integration, which is the only technique that seems feasible to me for brms models of arbitrary multilevel structure. 

Suppose we have a logistic regression with a varying intercept and a constant slope of `x`, then we need to compute `mu = integral g(b0 + u + b1 * x) f(u) du` where g is the response function (logistic function in this case) and f is the density of the varying intercepts u.

Monte-carlo integration to approximate mu could be done via 

```R
u <- rnorm(<large number>, 0, sd_u)
# or some other way to sample values of u
mu = mean(g(b0 + u + b1 * x))
```

This is computationally demanding since we have to do this marginalization for each posterior draw, but one may be able to vectorize most of the computation at the expense of increased RAM requirements.

**Comment by jackobailey** [comment]
You could adapt the code I submitted (and that you reposted in the issue above) to do this quite easily without having to rely on Monte Carlo integration. It would still be computationally expensive where the varying effects have a lot of levels and/or the data are large, but probably less so than Monte Carlo integration and easier to implement.

**Comment by paul-buerkner** [comment]
This has been implemented via emmeans support in PR #1177. One can now choose the varying effects to marginalize over by specifying the `re_formula` argument in `emmeans(brmsfit, ...)` accordingly.
